opendatahub-io/odh-dashboard
# Fix metrics submission issue and handle errors

Closes #1359 

## Description


1. Change the type of `privilegedAttribute`, `unprivilegedAttribute` and `favorableOutcome` to meet the trusty AI server requirements for data types.
2. Create error handlers to handle the 4xx error that comes from the trusty AI server. For example, when you input some invalid value, trusty AI server will give a 400 Bad Request error back, we can show it in the creation modal now.

<img width="1904" alt="Screenshot 2023-06-13 at 3 09 52 PM" src="https://github.com/opendatahub-io/odh-dashboard/assets/37624318/2f807b6f-1616-4acd-bba7-daf2ac60e571">

3. Fix the problem that users cannot successfully delete SPD metrics.
4. Check the session storage every time entering the bias charts and update them. Because when users delete one configuration, we didn't update the session storage for the chart selections, we need to reflect all the changes that the user made.

## How Has This Been Tested?



To test description 1:
Try to create/duplicate a configuration, using the valid attributes to see if you could successfully create one

To test description 2:
Try to create/duplicate a configuration, using the invalid attributes like string for the `privilegedAttribute` field to see if you can get the error alert after hitting the submit button

To test description 3:
Try to delete an SPD type configuration (You cannot delete that before because the context used the wrong delete function)

To test description 4:
Select to show a metric to display -> Go to the configuration page -> Delete that metric -> Go back to view the metrics chart -> Make sure the selection is removed and the chart is also removed

## Test Impact


TBD

## Request review criteria:
